extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -168,6 +169,55 @@ private function getTitleFromRequest(){
 	}
 
 	/**
+	 * @desc Returns resource type for URI
+	 *
+	 * @throws BadRequestApiException
+	 */
+	public function getResourceType() {
+		$uri = $this->request->getVal( self::PARAM_URI, NULL );
+		if ( empty( $uri ) ) {
+			throw new BadRequestApiException( 'You need to pass resource URI' );
+		}
+
+		// TODO Should we encode $uriWithoutPrefix?
+		// Should it be encoded by client?
+
+		if ( !empty( $this->wg->ArticlePath ) ) {
+			$uriPrefix = str_replace( '$1', '', $this->wg->ArticlePath );
+		} else {
+			$uriPrefix = '/wiki/';
+		}
+		$uriPrefixLength = strlen( $uriPrefix );
+
+		// Cut everything <= $urlPrefixLength
+		if ( $uriPrefix === substr( $uri, 0, strlen( $uriPrefix ) ) ) {
+			$uriWithoutPrefix = substr( $uri, $uriPrefixLength );
+		} else if ( substr( $uri, 0, 1 ) === '/' ) {
+			$uriWithoutPrefix = substr( $uri, 1 );
+		} else {
+			$uriWithoutPrefix = $uri;
+		}
+
+		// TODO Should we do this?
+		// Cut everything >= '?'
```

Comment:
I think so. Those are query params which shouldn't be connected to the resource which type we want get.